cosmos/models/Workflow.py

```diff
@@ -560,10 +562,21 @@ def _process_finished_tasks(jobmanager):
     for task in jobmanager.get_finished_tasks():
         if task.NOOP or task.exit_status == 0:
             task.status = TaskStatus.successful
-            yield task
+        elif task.scheduling_error:
```

Comment:
In terms of retries: I think we'd want to set the retry limit to allow some reasonable amount of time for the cluster to either fail over and recover (10–15 minutes?) and/or wait patiently for a human being to address whatever is going on (2–3 hours). It needn't resubmit repeatedly during that time. Maybe 3 retries separated by 5–10 minutes followed by a nasty-looking error message, then sleep 2 hours, then a final retry before death...

The current implementation pauses everything for 5–10 minutes, even status updates for other tasks, after the error is detected. (I figured it's probably not good for Cosmos to ask about other tasks if the cluster is in strange shape, and it also was the easiest way to implement it). I think we'll need a way to track these timeouts across Tasks, perhaps? Hmm.